clang/lib/Analysis/UnsafeBufferUsage.cpp

```diff
@@ -747,7 +747,7 @@ const Expr *extractExtentFromSubviewDataCall(ASTContext &Context,
 static bool hasIntegeralConstant(const Expr *E, uint64_t Val, ASTContext &Ctx) {
   Expr::EvalResult ER;
 
-  if (E->EvaluateAsConstantExpr(ER, Ctx)) {
+  if (E->EvaluateAsInt(ER, Ctx)) {
```

Comment:
According to https://github.com/llvm/llvm-project/pull/124022/files, the result of `EvaluateAsConstantExpr` may not necessarily be able to `getInt()` later.